projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js

```diff
@@ -36,6 +38,9 @@ export const settings = {
 			window?.Jetpack_LaunchpadSaveModal || {};
 		const isInsideSiteEditor = document.getElementById( 'site-editor' ) !== null;
 		const isInsidePostEditor = document.querySelector( '.block-editor' ) !== null;
+		const isNewsletter = siteIntentOption === 'newsletter';
+		const isNewsletterPostEditor = isNewsletter && isInsidePostEditor;
+		const isNewsletterSiteEditor = isNewsletter && isInsideSiteEditor;
```

Comment:
This may be a personal preference, I'll let others chime in, but I find the addition of new constants and the if / else if / else checks to be a bit heavy in a file that's already got a lot of them. I haven't tested, but would something like this work? It feels more readable to me. Maybe that's just me though :)

```diff
diff --git a/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js b/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
index d2296bcb82..5ecd5e885f 100644
--- a/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
+++ b/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
@@ -38,9 +38,6 @@ export const settings = {
 			window?.Jetpack_LaunchpadSaveModal || {};
 		const isInsideSiteEditor = document.getElementById( 'site-editor' ) !== null;
 		const isInsidePostEditor = document.querySelector( '.block-editor' ) !== null;
-		const isNewsletter = siteIntentOption === 'newsletter';
-		const isNewsletterPostEditor = isNewsletter && isInsidePostEditor;
-		const isNewsletterSiteEditor = isNewsletter && isInsideSiteEditor;
 		const prevHasNeverPublishedPostOption = useRef( hasNeverPublishedPostOption );
 
 		const siteFragment = getSiteFragment();
@@ -59,40 +56,36 @@ export const settings = {
 			} );
 
 		function getModalContent() {
-			let modalBody;
+			const modalContent = {
+				title: __( 'Great progress!', 'jetpack' ),
+				body: __(
+					'You are one step away from bringing your site to life. Check out the next steps that will help you to launch your site.',
+					'jetpack'
+				),
+				actionButtonHref: launchPadUrl,
+				actionButtonTracksEvent: 'jetpack_launchpad_save_modal_next_steps',
+				actionButtonText: __( 'Next Steps', 'jetpack' ),
+			};
 
-			if ( isNewsletter ) {
-				if ( isNewsletterPostEditor ) {
-					modalBody = __(
+			if ( siteIntentOption === 'newsletter' ) {
+				if ( isInsidePostEditor ) {
+					modalContent.title = __( 'Your first post is published!', 'jetpack' );
+					modalContent.body = __(
 						'Congratulations! You did it. View your post to see how it will look on your site.',
 						'jetpack'
 					);
-				} else if ( isNewsletterSiteEditor ) {
-					modalBody = __(
+					modalContent.actionButtonHref = postLink;
+					modalContent.actionButtonTracksEvent = 'jetpack_launchpad_save_modal_view_post';
+					modalContent.actionButtonText = __( 'View Post', 'jetpack' );
+				} else if ( isInsideSiteEditor ) {
+					modalContent.body = __(
 						'You are one step away from bringing your site to life. Check out the next steps that will help you to setup your newsletter.',
 						'jetpack'
 					);
 				}
-			} else {
-				modalBody = __(
-					'You are one step away from bringing your site to life. Check out the next steps that will help you to launch your site.',
-					'jetpack'
-				);
 			}
 
-			return {
-				title: isNewsletterPostEditor
-					? __( 'Your first post is published!', 'jetpack' )
-					: __( 'Great progress!', 'jetpack', /* dummy arg to avoid bad minification */ 0 ),
-				body: modalBody,
-				actionButtonHref: isNewsletterPostEditor ? postLink : launchPadUrl,
-				actionButtonTracksEvent: isNewsletterPostEditor
-					? 'jetpack_launchpad_save_modal_view_post'
-					: 'jetpack_launchpad_save_modal_next_steps',
-				actionButtonText: isNewsletterPostEditor
-					? __( 'View Post', 'jetpack' )
-					: __( 'Next Steps', 'jetpack', /* dummy arg to avoid bad minification */ 0 ),
-			};
+			return modalContent;
 		}
 
 		useEffect( () => {

```